### ISSUE TITLE
Metadata for annotate the type of a groundtruth-dataset

### ISSUE THREAD

**Issue by tboenig** [OCR-D/spec#165]
from Uwe Hartwig @M3ssman
Hello @all! I'd like to annotate the type of a groundtruth-dataset in PAGE to categorize them later as article-page, announcement-page, timetable-pages for trains and alike, maybe with some more differentiates (1-colum, 2-colum, ...). Is this a valid use-case for PAGE XML? Which element or element-attribute fits best? Something like /PageType/Labels@value ? Thers's a type-attribute for the Page element, but this is fixed to common structure type elements like cover, index or title-page ...

**Comment by tboenig** [comment]
Hallo @M3ssmann,

which different TYPES, do you have?
- `announcement-page`, 
-` timetable-pages for trains`

That`s very special Types. 

Do you really want to store the information in Page XML?
Is this mandatory for the application? Does the information have to be at the region/line/word?

**Comment by M3ssman** [comment]
@tboenig 
No, we ~~are~~ try to to differentiate between relative simple newspaper layouts and really tricky ones for announcements and table-like data (timetables, stock news). 

We are using this at our groundtruth frame level. "Groundtruth Frame" means, that we almost never transcript a whole page as groundtruth. Most times we use about the half of a page. This decision was made to handle a broader spectrum of pages, rather than to concentrate on a small number of items.

Having this would be fine, so one could quite easy pick hard layouts for further investigations just from metadata without taking a look to the actual scans. I utilized the OtherTag Elements from ALTO V3 this way, but since we switched to PAGE (PAGE 2013), I didn't find a differrnt way to keep this information than to store it in the filenames. So we go with gt-files like: `1681877805_J_0075_0001.gt.art1.xml`, where `art1` means: 1 column from an article page.

Currently in our case it's a mix of semantic + layout information. It would be really, really nice to be able to express on page level something like `article with 3 columns` or  `annoucement with mainly 8 columns` (since ads and announcements are sometimes mixed in virtually impossible ways)

**Comment by bertsky** [comment]
> We are using this at our groundtruth frame level. "Groundtruth Frame" means, that we almost never transcript a whole page as groundtruth. Most times we use about the half of a page. This decision was made to handle a broader spectrum of pages, rather than to concentrate on a small number of items.

A wise decision IMO!

> but since we switched to PAGE (PAGE 2013), I didn't find a differrnt way to keep this information than to store it in the filenames

Since 2013, PAGE has `Page/@type` and `@custom` everywhere, and since 2018 there's `Label`. This should suffice...

> So we go with gt-files like: `1681877805_J_0075_0001.gt.art1.xml`, where `art1` means: 1 column from an article page.

There are plans to integrate IIIF support into OCR-D. Then you could keep the reference to the original full page, while processing (or training) only sections with OCR-D pipelines.

Assuming PAGE-XML is used only for such newspaper page sections (regardless of whether their original is represented/linked somewhere), indeed the `PageTypeSimpleType` vocabulary is too restrictive, but you could use `Page/Labels/Label` for this (even linking to an external schema) AFAICT.

> like `article with 3 columns` or `annoucement with mainly 8 columns`

I assume you don't mean this literally (as a compound type), but that one could quickly query how many columns a "page" (i.e. section) has. PAGE does not have a dedicated representation for text columns, but uses `TextRegion` recursion. So you could e.g. introduce a convention of annotating `TextRegion[@type="other",@custom="subtype:column"]` which then has other regions (text and non-text) inside. Your query would then need to use complex XPath expressions, but it's possible. (Maybe these queries could also be precomputed and stored at `Page/@custom` or even in the METS.)

**Comment by tboenig** [comment]
First of all, thank you very much. Now I can better imagine the problem with the columns.

The proposal
`TextRegion[@type="other",@custom="subtype:column"]`

leads us in the right direction, but:

my suggestion:
`TextRegion[@type="paragraph",@custom="#column"]`

@type="other"
should be used for regions that are not a paragraph....
A column can contain one or more paragraphs or other things that need to be specifically named.

Recursion is also my solution at the moment.

I send an example.

We haven't solved the specific semantic things yet: like announcements, advertisements and other special things.

**Comment by tboenig** [comment]
[hohberg_georgica01_1682_0601.zip](https://github.com/OCR-D/spec/files/4805722/hohberg_georgica01_1682_0601.zip)

**Comment by bertsky** [comment]
@tboenig 
> The proposal
> `TextRegion[@type="other",@custom="subtype:column"]`
> 
> leads us in the right direction, but:
> 
> my suggestion:
> `TextRegion[@type="paragraph",@custom="#column"]`
> 
> @type="other"
> should be used for regions that are not a paragraph....

That's exactly what columns are: regions that are not a paragraph (themselves)!

Your example is a perfect illustration of that. The top-level recursive text regions representing columns (regardless of whether you use `@custom="#column"` or some other formatting) _should_ have `@type="other"`.

> A column can contain one or more paragraphs or other things that need to be specifically named.

Just what I said above.

> We haven't solved the specific semantic things yet: like announcements, advertisements and other special things.

Oh, but we have, havent' we? 

> > but you could use `Page/Labels/Label` for this (even linking to an external schema) AFAICT.
